pkg/build/graph/analysis/bc.go

```diff
@@ -68,6 +71,86 @@ bc:
 	return markers
 }
 
+// FindMissingInputImageStreams checks all build configs and confirms that their From element exists
+func FindMissingInputImageStreams(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+OUTER:
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, istNode := range g.PredecessorNodesByEdgeKind(bcNode, buildedges.BuildInputImageEdgeKind) {
+			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
+				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+
+				// note, BuildConfig.Spec.BuildSpec.Strategy.[Docker|Source|Custom]Stragegy.From Input of ImageStream type will get converted to ImageStreamTag on the vX to api conversion
+
+				if !imageStreamNode.IsFound {
+					markers = append(markers, osgraph.Marker{
+						Node:         bcNode,
+						RelatedNodes: []graph.Node{istNode},
+
+						Severity: osgraph.ErrorSeverity,
+						Key:      MissingInputImageStreamErr,
+						Message: fmt.Sprintf("%s builds from %s, but the image stream does not exist.",
+							bcNode.(*buildgraph.BuildConfigNode).ResourceString(), imageStreamNode.ResourceString()),
+						Suggestion: osgraph.Suggestion(fmt.Sprintf("create the image stream %s with the required images", imageStreamNode.ResourceString())),
+					})
+
+					continue OUTER
+				}
+
+				if tagNode, ok := istNode.(*imagegraph.ImageStreamTagNode); ok {
+					imageStream := imageStreamNode.Object().(*imageapi.ImageStream)
+					if _, ok := imageStream.Status.Tags[tagNode.ImageTag()]; !ok {
+						markers = append(markers, osgraph.Marker{
+							Node: bcNode,
+							RelatedNodes: []graph.Node{istNode,
+								imageStreamNode},
+
+							Severity: osgraph.WarningSeverity,
+							Key:      MissingImageStreamTagWarning,
+							Message: fmt.Sprintf("%s builds from %s, but the image stream tag does not exist.",
+								bcNode.(*buildgraph.BuildConfigNode).ResourceString(), istNode.(*imagegraph.ImageStreamTagNode).ResourceString()),
+							Suggestion: osgraph.Suggestion(fmt.Sprintf("examine analysis of build config outputs from this command and see if they build %s", istNode.(*imagegraph.ImageStreamTagNode).ResourceString())),
+						})
+					}
+					continue OUTER
+				}
+
+				if tagNode, ok := istNode.(*imagegraph.ImageStreamImageNode); ok {
+					imageStream := imageStreamNode.Object().(*imageapi.ImageStream)
+					// cycle through the imageStream.Status.Tags.[]TagEvent.DockerImageReference and see if the image id is present
+					var imageID string
+					for _, tagEventList := range imageStream.Status.Tags {
```

Comment:
part of next push ... also, in looking at this, I see the imageID used as input into the marker definition was not getting set